### PR TITLE
Unified Linux makefiles (x86/x64)

### PR DESCRIPTION
**Pull request by Holzhaus**
This PR merges the lin86/lin64 Makefiles into one Makefile.

Buildarch is auto-detected via `uname -m`, but can also be enforced (which might be useful for cross-compiling):

```
make ARCH=i686
```

Compilation should work as soon as #702 has been fixed (that error occurs also with old makefiles after you replace `HOST_NO_AREC` with `TARGET_NO_AREC`).

### CODE REVIEW

**Comment by skmp** [comment]
Hmm, can you make it print a warning and ask yes/no if x64 build is done? x86 build is more than 3 times faster right now.

**Comment by skmp** [comment]
Also, maybe a linux-intel folder makes more sense? Since this doesn't handle arm and it's impossible variations

**Comment by Holzhaus** [comment]
> Also, maybe a linux-intel folder makes more sense? Since this doesn't handle arm and it's impossible variations

Currently yes. But I'd like to merge in the Rapi2, beagle, etc stuff, too (at some point at least).

**Comment by Holzhaus** [comment]
OK, this works now. PR #705 also merges the other Linux makefiles.

File: shell/linux/Makefile [hunk 1]
```diff
@@ -2,18 +2,41 @@
...
 
-RZDCY_SRC_DIR = ../../core
+MFLAGS := 
+ASFLAGS := 
+LDFLAGS :=
+CFLAGS := 
+CXXFLAGS :=
 
-include $(RZDCY_SRC_DIR)/core.mk
+ifeq (,$(ARCH))
```
Comment (skmp) [inline shell/linux/Makefile]: Wouldnt ?= also work?

**Pull request merged**